lib/gcloud/search/result.rb

```diff
@@ -0,0 +1,61 @@
+#--
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud/search/result/list"
+
+module Gcloud
+  module Search
+    ##
+    # = Result
+    #
+    # See Gcloud#search
+    class Result
+      ##
+      # Creates a new Result instance.
+      def initialize raw #:nodoc:
+        @raw = raw
```

Comment:
In Document `@raw` is set to an empty hash in the initializer, and the way to convert to/from hash is using `Document.from_hash` and `Document#to_hash`. I think Result and Document should behave the same way.